app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala

```diff
@@ -340,51 +326,52 @@ class MovementsRepositoryImpl @Inject() (
       movementMRNFilter(movementReferenceNumber),
       movementLRNFilter(localReferenceNumber)
     )
-    val messageSummaryQuery = Json.obj("$ifNull" -> Json.arr("$movementSummary", "[]"))
-
     val (from, itemCount) = indices(page, count)
-
-    val aggregates = Seq(
-      Aggregates.filter(selector),
+    val filterAggregates  = Seq(Aggregates.filter(selector))
+    val aggregates = filterAggregates ++ Seq(
       Aggregates.sort(descending("updated")),
-      Aggregates.project(Projections.exclude("messages")),
-      Aggregates.facet(
-        Facet("totalCount", Aggregates.count()),
-        Facet("movementSummary", Aggregates.skip(from), Aggregates.limit(itemCount))
-      ),
-      Aggregates.project(Document("totalCount" -> Codecs.toBson(totalCountQuery()), "movementSummary" -> Codecs.toBson(messageSummaryQuery)))
+      Aggregates.skip(from),
+      Aggregates.limit(itemCount)
     )
+    for {
+      perPageMovements <- filterPerPageMovements(aggregates)
+      totalCount       <- totalCountMovements(filterAggregates)
+    } yield PaginationMovementSummary(TotalCount(totalCount), perPageMovements)
+
+  }
 
-    mongoRetry(Try(collection.aggregate[PaginationMovementSummary](aggregates)) match {
+  private def filterPerPageMovements(aggregates: Seq[Bson]): EitherT[Future, MongoError, Vector[MovementWithoutMessages]] =
+    mongoRetry(Try(collection.aggregate[MovementWithoutMessages](aggregates)) match {
```

Comment:
This method duplicates `filterPerPageMessages` apart from the return type -- rather than duplicate code, use a generic:

```
private def filterPerPage[R](aggregates: Seq[Bson]): EitherT[Future, MongoError, Vector[R]]
```

It might need an implicit defining too.